### ISSUE TITLE
Strange LIMIT behavior with multiple joins/subqueries

### ISSUE THREAD

**Issue by SneakyMax** [crate/crate#6641]
**CrateDB version**: 2.2.3

**JVM version**: `Java(TM) SE Runtime Environment (build 1.8.0_144-b01)`

**OS version / environment description**: Ubuntu 16.04

Here's the large query I have:

```
SELECT agg_total,
       agg_unique,
       agg_median,
       agg_sum,
       id,
       title,
       table_B._id AS internal_id
FROM
(
    SELECT col_A,
           percentile(col_E, 0.5) AS agg_median,
           SUM(col_E) AS agg_sum
    FROM table_A
    WHERE col_B = 'value'
          AND col_C IS NULL
          AND col_D != 'Unknown'
          AND col_E IS NOT NULL
          AND col_F = 'value_1'
    GROUP BY col_A
) group_A
INNER JOIN (
    SELECT COUNT(*) AS agg_total,
           COUNT(DISTINCT amplitude_id) AS agg_unique,
           col_A
    FROM table_A
    WHERE col_B = 'value'
          AND col_C IS NULL
          AND col_D != 'Unknown'
          AND col_F = 'value_2'
    GROUP BY col_A
) group_B ON group_A.col_A = group_B.col_A
INNER JOIN table_B ON group_A.col_A = table_B._id
WHERE agg_total >= 0
      AND type = 'type_1'
      AND "exists" = true
ORDER BY agg_unique DESC
LIMIT X;
```

And here's the strange behavior:

- If the `LIMIT` is 100, this query returns 3 rows (the max)
- If the `LIMIT` is 25, this query returns 1 row
- If the `LIMIT` is 2, this query returns no rows
- If the first `INNER JOIN` is changed to a `FULL OUTER JOIN`, the `LIMIT` behaves as expected (LIMIT 1 = 1 row, LIMIT 2 = 2 rows, LIMIT 3+ = 3 rows)

Am I just crazy and don't understand SQL? I would assume the LIMIT would limit the number of final rows returned, but it's almost like the LIMIT is running...inside the subqueries?

I tried running an EXPLAIN and I don't really understand the output. I can email it to you if you'd like.

**Comment by matriv** [comment]
@SneakyMax can you please email me the outputs of explain for both INNER JOIN and FULL OUTER JOIN? marios [ at ] crate.io

**Comment by matriv** [comment]
@SneakyMax Would also be helpful if you could provide the table schemas and sample data.

**Comment by SneakyMax** [comment]
On it

**Comment by matriv** [comment]
@SneakyMax It's actually a bug in our `2.2.x` versions. In `2.3` it doesn't occur.
Thanks for reporting, for the info you shared and for your patience ;-)
The fix will be available with the next `2.2` hotfix release.
